.changeset/nasty-tigers-sip.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+Add "rawContent" helper to imported markdown files. This exposes both the raw markdown content when using rawContent() and the parsed HTML when using rawContent.html()
```

Comment:
so naming: `rawContent.html()` is really confusing because I would think that I could use its output as HTML. I don't think docs are enough here given how explicit `html()` sounds like html, I would never think to even check the docs. 

Can we consider a different name? 
- `output()`
- `compile()`
- `render()`
- `renderWithJSX()`
- `renderToAstro()`

Reply:
Hm, the output IS html actually,  not the JSX transform. This is the output with markdown compiled to HTML and JSX expressions _ignored,_ so I don’t think those suggested name changes would work here. See discord bikeshedding discussion [starting around here](https://discord.com/channels/830184174198718474/845430950191038464/979108718836846673).

Reply:
Ah, sorry I missed that! I still stand by the concern, where `html()` would indicate to me that the result is valid HTML. JSX-like expressions would break my expectations, as would un-rendered Astro-components like `<MyButton foo={42} />`.

Reply:
Well, it is still valid html, so we didn’t think it was a huge concern after discussing with @delucis. The only other option we found was “htmlWithUnparsedJsx,” which isn’t a very pleasant name. Is this what you’d prefer?

Reply:
Yup, less about literal correctness and more about the fact that this is a really unintuitive error that we'd be introducing: `<article set:html={post.rawContent().html()} />`

Also while we're on the topic, I'd recommend avoiding double-dipping a function as an object, for `rawContent()` and `rawContent.html()`. Again, it's literally correct but much harder to document and reason about than an object like `post.content = {raw, compiled}`.

Reply:
If you're asking for a concrete proposal from me, I'd vote for this:

```
await post.content.raw()      // returns the raw body of the file, minus markdown
await post.content.compiled() // returns the markdown compiled to valid astro syntax
```